### PR TITLE
r/log_analytics_workspace - Expose `capacity_reservation_level` and `CapacityReservation` SKU value

### LINKED ISSUE #10553: Support for capacityReservationLevel in azurerm_log_analytics_workspace
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
Azure Log Analytics Workspaces have an option for capacityReservationLevel. 

This option is not surfaced in AzureRM Terraform module

### New or Affected Resource(s)

* azurerm_log_analytics_workspace

### Potential Terraform Configuration

```hcl
resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "East US"
}

resource "azurerm_log_analytics_workspace" "example" {
  name                = "acctest-01"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  sku                 = "CapacityReservation"
  capacityReservationLevel = 100
  retention_in_days   = 30
}
```

### References

Supported in Azure ARM template: https://docs.microsoft.com/en-us/azure/templates/microsoft.operationalinsights/2020-03-01-preview/workspaces 
* #0000

### PR DESCRIPTION
**Pull request by brunhil**
Fixes #10553 

=== RUN   TestAccLogAnalyticsWorkspace_withCapacityReservation
=== PAUSE TestAccLogAnalyticsWorkspace_withCapacityReservation
=== CONT  TestAccLogAnalyticsWorkspace_withCapacityReservation
--- PASS: TestAccLogAnalyticsWorkspace_withCapacityReservation (152.36s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/loganalytics        154.175s

### CODE REVIEW

File: azurerm/internal/services/loganalytics/log_analytics_workspace_resource.go [hunk 1]
```diff
@@ -84,11 +84,18 @@ func resourceLogAnalyticsWorkspace() *schema.Resource {
 					string(operationalinsights.WorkspaceSkuNameEnumPremium),
 					string(operationalinsights.WorkspaceSkuNameEnumStandalone),
 					string(operationalinsights.WorkspaceSkuNameEnumStandard),
+					string(operationalinsights.WorkspaceSkuNameEnumCapacityReservation),
 					"Unlimited", // TODO check if this is actually no longer valid, removed in v28.0.0 of the SDK
 				}, true),
 				DiffSuppressFunc: logAnalyticsLinkedServiceSkuChangeCaseDifference,
 			},
 
+			"capacity_reservation_level": {
```
Comment (katbyte) [inline azurerm/internal/services/loganalytics/log_analytics_workspace_resource.go:93]: the use of "level" seems vague here, going by https://docs.microsoft.com/en-us/azure/azure-monitor/platform/manage-cost-storage it seems to corraspond to "gb/d" so would this be better as
```suggestion
			"reservation_capcity_in_gb_per_day": {
```
or even
```suggestion
			"reservation_capcity": {
```
WDYT?

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks for this PR @brunhil, looks good but i do have a question about the name again, looking up what level meant i wonder if it would be better to change it as per my comments left inline

File: azurerm/internal/services/loganalytics/log_analytics_workspace_resource.go (hunk 1 above)
Comment (brunhil) [reply azurerm/internal/services/loganalytics/log_analytics_workspace_resource.go:93]: I was torn with `capacity_reservation_level` on first implementation and couldn't come up with a better one at the time. One of the hardest things to do in programming is properly name things 😅 .

That being said I think your suggestion of `reservation_capcity_in_gb_per_day` is a perfect balance. It simplifies it but allows users to not have to pull up the documentation to find out that it is in GB per day 😄 .

**Review by jackofallops — APPROVED** [review]
Thanks @brunhil - LGTM now 👍

**Comment by jackofallops** [comment]
Tests Passing:
image

**Pull request merged**